symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/stream/StreamService.java

```diff
@@ -0,0 +1,288 @@
+   */
+  public RoomDetail setRoomActive(String roomId, Boolean active) {
+    try {
+      return streamsApi.v1RoomIdSetActivePost(roomId, active, authSession.getSessionToken());
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Update the attributes of an existing chatroom.
+   *
+   * @param roomId The id of the room to be updated
+   * @param roomAttributes The attributes to be updated to the room
+   * @return The information of the room after being updated.
+   * @see <a href="https://developers.symphony.com/restapi/reference#update-room-v3">Update Room V3</a>
+   */
+  public V3RoomDetail updateRoom(String roomId, V3RoomAttributes roomAttributes) {
+    try {
+      return streamsApi.v3RoomIdUpdatePost(roomId, authSession.getSessionToken(), roomAttributes);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Retrieve a list of all streams of which the requesting user is a member,
+   * sorted by creation date (ascending).
+   *
+   * @param filter The stream searching criteria
+   * @return The list of streams retrieved according to the searching criteria.
+   * @see <a href="https://developers.symphony.com/restapi/reference#list-user-streams">List Streams</a>
+   */
+  public List<StreamAttributes> listStreams(StreamFilter filter) {
+    try {
+      return streamsApi.v1StreamsListPost(authSession.getSessionToken(), null, null, filter);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Get information about a particular stream.
+   *
+   * @param streamId The stream id
+   * @return The information about the stream with the given id.
+   * @see <a href="https://developers.symphony.com/restapi/reference#stream-info-v2">Stream Info V2</a>
+   */
+  public V2StreamAttributes getStreamInfo(String streamId) {
+    try {
+      return streamsApi.v2StreamsSidInfoGet(streamId, authSession.getSessionToken());
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * List attachments in a particular stream.
+   *
+   * @param streamId The stream id
+   * @param sinceInMillis Timestamp in millis of first required attachment
+   * @param toInMillis Timestamp in millis of last required attachment
+   * @param sort Attachment date sort direction : ASC or DESC (default to ASC)
+   * @return List of attachments in the stream with the given stream id.
+   * @see <a href="https://developers.symphony.com/restapi/reference#list-attachments">List Attachments</a>
+   */
+  public List<StreamAttachmentItem> getAttachmentsOfStream(String streamId, Long sinceInMillis, Long toInMillis, AttachmentSort sort) {
+    try {
+      return streamsApi.v1StreamsSidAttachmentsGet(streamId, authSession.getSessionToken(), sinceInMillis, toInMillis, null, sort.name());
```

Comment:
Should this be put in the StreamService or in the MessageService ? This is put in the messages part of the documentation : https://developers.symphony.com/restapi/reference#list-attachments

Reply:
I'm also not sure with this method because this api is marked with tag `Streams` in the `symphony-api-spec`, so it is generated within `StreamsApi` class.